src/RockBridge.cpp

```diff
@@ -89,18 +89,22 @@ void RockBridge::worldCreated(std::string const& worldName)
         // Create and initialize one rock component for each gazebo model
 
         sdf::ElementPtr modelSDF = (*model_it)->GetSDF();
-        if(modelSDF->HasElement("plugin"))
+        if (modelSDF->HasElement("plugin"))
         {
             sdf::ElementPtr pluginElement = modelSDF->GetElement("plugin");
-            std::string pluginFilename = pluginElement->Get<std::string>("filename");
-            gzmsg <<"RockBridge: found plugin "<< pluginFilename << std::endl;
-            if(pluginFilename == "libgazebo_thruster.so")
-            {
-                ThrusterTask* model_task = new ThrusterTask();
-                model_task->setGazeboModel(world, (*model_it) );
-                setupTaskActivity(model_task);
+            while (pluginElement) {
+                std::string pluginFilename = pluginElement->Get<std::string>("filename");
+                gzmsg <<"RockBridge: found plugin "<< pluginFilename << std::endl;
+                if (pluginFilename == "libgazebo_thruster.so")
+                {
+                    ThrusterTask* thruster_task = new ThrusterTask();
+                    thruster_task->setGazeboModel(world, (*model_it), pluginElement);
+                    setupTaskActivity(thruster_task);
+                }
+                pluginElement = pluginElement->GetNextElement("plugin");
```

Comment:
Do we have to create a plugin for each thruster we create ? 
For instance, in the flat_fish model we would create 6 plugins inside it !?